pkg/prometheus/operator.go

```diff
@@ -1985,17 +2047,38 @@ func (c *Operator) selectPodMonitors(p *monitoringv1.Prometheus) (map[string]*mo
 
 	level.Debug(c.logger).Log("msg", "filtering namespaces to select PodMonitors from", "namespaces", strings.Join(namespaces, ","), "namespace", p.Namespace, "prometheus", p.Name)
 
-	podMonitors := []string{}
 	for _, ns := range namespaces {
 		cache.ListAllByNamespace(c.pmonInf.GetIndexer(), ns, podMonSelector, func(obj interface{}) {
 			k, ok := c.keyFunc(obj)
 			if ok {
 				res[k] = obj.(*monitoringv1.PodMonitor)
-				podMonitors = append(podMonitors, k)
 			}
 		})
 	}
 
+	// If denied by Prometheus spec, filter out all pod monitors that access
+	// the file system.
+	if p.Spec.ArbitraryFSAccessThroughSMs.Deny {
```

Comment:
As this field was named after ServiceMonitors, should we consider renaming it before using it with PodMonitors?

Reply:
I think we should just not allow arbitrary filesystem access at all, and rather focus on implementing safe functionality for the same use cases (the only reason I've ever seen this used for is serviceaccount tokens).

Reply:
fixed by removing configuration options accessing the filesystem